actionmailer/test/message_delivery_test.rb

```diff
@@ -93,4 +98,12 @@ def test_should_enqueue_and_run_correctly_in_activejob
       @mail.deliver_later(queue: :another_queue)
     end
   end
+
+  test "can override base job class" do
+    assert_performed_with(job: RescueJob, args: ['DelayedMailer', 'test_message', 'deliver_now', 1, 2, 3], queue: "another_queue") do
+      ActionMailer::Base.delivery_job_class = RescueJob
+      @mail.deliver_later(queue: :another_queue)
+      ActionMailer::Base.delivery_job_class = ActionMailer::DeliveryJob
```

Comment:
Can we assign the original value back. Also we should do that in an ensure block.

```ruby
begin
  original_delivery_job_class = ActionMailer::Base.delivery_job_class
  ActionMailer::Base.delivery_job_class = RescueJob
  # ...
ensure
  ActionMailer::Base.delivery_job_class = original_delivery_job_class
end
```